tests/Doctrine/Tests/ORM/Functional/EnumTest.php

```diff
@@ -347,15 +349,10 @@ public function testEnumWithNonMatchingDatabaseValueThrowsException(string $card
             [$metadata->fieldMappings['id']['columnName'] => $card->id]
         );
 
-        $this->expectException(MappingException::class);
```

Comment:
Why do we need to change the expectation here? This could qualify as a breaking change if calling code expects a `MappingException` in this scenario.

Reply:
@derrabus Before the change the string was converted to an enum in the `ReflectionEnumProperty` class https://github.com/doctrine/orm/blob/1ed0057276b3846ba6f1c3e77f926bb69467a033/lib/Doctrine/ORM/Mapping/ReflectionEnumProperty.php#L88-L108 
Now it's converted earlier in the `AbstractHydrator` class instead
https://github.com/doctrine/orm/blob/1ed0057276b3846ba6f1c3e77f926bb69467a033/lib/Doctrine/ORM/Internal/Hydration/AbstractHydrator.php#L700-L709

Since the `AbstractHydrator` class throws a different exception the test fails. I'm not sure if this inconsistency in exceptions is a bug itself.

Reply:
@HypeMC we should wrap with the same try-catch block and throw MappingException I think

Reply:
@HypeMC can you please incorporate this change?

Reply:
Yes, I've already started, unfortunately it's not as trivial as I thought. The current exception message contains the class name, which is currently not always available. I'll try to finish the PR in the following days.

Reply:
replace 
https://github.com/doctrine/orm/blob/b41f5ec60530c87b48fc5ccd2c93729bc8060f64/lib/Doctrine/ORM/Internal/Hydration/SimpleObjectHydrator.php#L143-L145 
with
```php
            if ($value !== null && isset($cacheKeyInfo['enumType'])) {
                try {
                    $value = $this->buildEnum($value, $cacheKeyInfo['enumType']);
                } catch (ValueError $e) {
                    throw MappingException::invalidEnumValue(
                        $entityName,
                        $cacheKeyInfo['fieldName'],
                        (string) $value,
                        $cacheKeyInfo['enumType'],
                        $e
                    );
                }
            }
```

One issue though - the `AbstractHydrator::hydrateRowData()` throws only `HydrationException`, so throwing `MappingException` from here will result in psalm errors. @derrabus what do you suggest, should we update the `@throws` PHPDoc or should we wrap the Mapping exception into HydrationException (this by the same logic could be considered BC break)

Reply:
@michnovka Good point, I wanted to change the `buildEnum` method to throw a `MappingException`, but that would actually also be a BC break, so this makes the most sense.